pkg/ibm_plugin/server/plugin_integration_test.go

```diff
@@ -0,0 +1,406 @@
+	orchestrator.Setup(orchestratorServerConfig, true)
+
+	// start ibm plugin server
+	fmt.Println("Setting up IBM server")
+	ibmServer := Setup(IBMServerPort, orchestratorServerAddr)
+
+	fmt.Println("Setting up kv store server")
+	tagging.Setup(dbPort, taggingPort, true)
+
+	fmt.Println("Setting up kv tagging server")
+	kvstore.Setup(dbPort, kvstorePort, true)
+
+	// Create IBM VM
+	fmt.Println("\nCreating IBM VM...")
+	imageIdentity := vpcv1.ImageIdentityByID{ID: &image}
+	zoneIdentity := vpcv1.ZoneIdentityByName{Name: &zone}
+	myTestProfile := string(testProfile)
+
+	testPrototype := &vpcv1.InstancePrototypeInstanceByImage{
+		Image:   &imageIdentity,
+		Zone:    &zoneIdentity,
+		Name:    core.StringPtr(instanceName),
+		Profile: &vpcv1.InstanceProfileIdentityByName{Name: &myTestProfile},
+	}
+
+	description, err := json.Marshal(vpcv1.CreateInstanceOptions{InstancePrototype: vpcv1.InstancePrototypeIntf(testPrototype)})
+	require.NoError(t, err)
+
+	resource := &paragliderpb.CreateResourceRequest{Name: instanceName, Deployment: &paragliderpb.ParagliderDeployment{Id: testDeployment, Namespace: testNamespace}, Description: description}
+	res, err := ibmServer.CreateResource(context.Background(), resource)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	// append instance's ID
+	URIParts := strings.Split(res.Uri, "/")
+	resID := IBMResourceIDPrefix + URIParts[len(URIParts)-1]
+
+	// Add permit list for IBM VM
+	fmt.Println("Adding IBM permit list rules...")
+
+	addRulesRequest := &paragliderpb.AddPermitListRulesRequest{
+		Namespace: testNamespace,
+		Resource:  resID,
+		Rules:     pingTestPermitList,
+	}
+
+	resp, err := ibmServer.AddPermitListRules(context.Background(), addRulesRequest)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	utils.Log.Printf("Test response: %+v", resp)
+}
+
+// TODO(cohen-j-omer) will add IBM connectivity check method
+// usage: go test --tags=ibm -run TestMulticloudIBMAzure -timeout 0
+// -timeout 0 removes limit of 10 minutes runtime, which is necessary due to long deployment time of Azure's VPN.
+func TestMulticloudIBMAzure(t *testing.T) {
+	dbPort := 6379
+	kvstorePort := 7993
+	taggingPort := 7994
+	// ibm config
+	IBMServerPort := 7992
+	IBMResourceIDPrefix := testResourceIDUSEast1
+	image, zone, instanceName := testImageUSEast, testZoneUSEast1, testInstanceNameUSEast1
+	// azure config
+	azureServerPort := 7991
+	azureSubscriptionId := azure.GetAzureSubscriptionId()
+
+	// removes all of paraglide's deployments on IBM
+	region, err := ibmCommon.ZoneToRegion(zone)
+	require.NoError(t, err)
+	defer sdk.TerminateParagilderDeployments(resourceGroupID, region)
+
+	// requires resource group creation/deletion privileges
+	if DoesHaveResourceGroupPrivileges {
+		azureResourceGroupName := azure.SetupAzureTesting(azureSubscriptionId, "ibmazure")
+		defer azure.TeardownAzureTesting(azureSubscriptionId, azureResourceGroupName)
+	}
```

Comment:
This should be fixed once #290 is merged. Leaving this comment so we know to change this once that happens.